physics/rrtmgp_aerosol_optics.F90

```diff
@@ -65,34 +57,37 @@ subroutine rrtmgp_aerosol_optics_run(doSWrad, doLWrad, nCol, nLev, nTracer, nTra
     ! Outputs
     real(kind_phys), dimension(:,:), intent(out) :: &
          aerodp                   ! Vertical integrated optical depth for various aerosol species 
-    type(ty_optical_props_2str),intent(out) :: &
-         sw_optical_props_aerosol ! RRTMGP DDT: Longwave aerosol optical properties (tau)
-    type(ty_optical_props_1scl),intent(inout) :: &
-         lw_optical_props_aerosol ! RRTMGP DDT: Longwave aerosol optical properties (tau)
+    real(kind_phys), dimension(:,:,:), intent(out) :: &
+         aerlw_tau,             & ! Longwave aerosol optical depth
+         aerlw_ssa,             & ! Longwave aerosol single scattering albedo
+         aerlw_g,               & ! Longwave aerosol asymmetry parameter
+         aersw_tau,             & ! Shortwave aerosol optical depth 
+         aersw_ssa,             & ! Shortwave aerosol single scattering albedo
+         aersw_g                  ! Shortwave aerosol asymmetry parameter
     integer, intent(out) :: &
          errflg                   ! CCPP error flag
     character(len=*), intent(out) :: &
          errmsg                   ! CCPP error message
 
     ! Local variables
-    real(kind_phys), dimension(nCol, nLev, lw_gas_props%get_nband(), NF_AELW) :: &
+    real(kind_phys), dimension(nCol, nLev, lw_gas_props%get_nband(), 3) :: &
```

Comment:
seems a little weird to be going from NF_AELW to a hard-coded constant. I guess it will never change?

Reply:
@grantfirl In this context 3 is fixed. The third dimension indexes different fields in a stacked array, optical-depth, single-scattering albedo, and asymmetry parameter